lib/pathfinder/PathfindingRules.cpp

```diff
@@ -62,7 +62,7 @@ void MovementCostRule::process(
 
 		const int destinationLayerMaxMovePoints = pathfinderHelper->getMaxMovePoints(destination.node->layer);
 		const float costBeforeConversion = static_cast<float>(destMovePointsLeft) / sourceLayerMaxMovePoints;
-		const float costAfterConversion = static_cast<float>(movementPointsAfterEmbark) / destinationLayerMaxMovePoints;
+		auto costAfterConversion = static_cast<float>(movementPointsAfterEmbark) / destinationLayerMaxMovePoints;
```

Comment:
If we change one, why not changing the others? Also Will variable be const after this change? Is it important for it to be const?

Reply:
No idea why only one line got changed.

Variable is indeed no longer const, but should have no effect on code.

Reply:
I'll check why the scripts ignored ``costBeforeConversion`.

Since it's recommended to make any variable a const if possible, I'll go through the changes and re-add it.

Reply:
To be honest might be a good idea to exclude float's/int's/etc from this change after all.
for example this code will turn into
```cpp
const int destinationLayerMaxMovePoints = ...
const auto costBeforeConversion = ...
const auto costAfterConversion = ...
```
So in order to check type of `destinationLayerMaxMovePoints` you need to look left, but in order to check type of other two you need to look at cast. And what's worse is that in this case we have static_cast not on whole expression, but only on one operand.

Reply:
I'll split it up in several PRs so it's easier to keep an overview.

I've just replaced the types in initializations using `new`.